src/index.js

```diff
@@ -57,17 +62,23 @@ export default function alias(options = {}) {
       const importerId = normalizeId(importer);
 
       // First match is supposed to be the correct one
-      const toReplace = aliasKeys.find(key => matches(key, importeeId));
-
-      if (!toReplace || !importerId) {
+      const matchedEntry = entries.find(entry => matches(entry.find, importeeId, entry.isRegEx));
+      if (!matchedEntry || !importerId) {
         return null;
       }
 
-      const entry = options[toReplace];
+      const toReplace = matchedEntry.find;
+      const isDir = matchedEntry.isRegEx && toReplace.source
```

Comment:
I do not really understand why this exists? It is not documented in the Readme, causes a lot of additional friction in the code and only seems to be triggered in the first test anyway, where I am still not sure what the intended effect is?

Reply:
absolutely correct - that thing was carelessly implemented to fix my own stupidity in simply providing an adequate replacement value ... hence I've simplyfied the code and removed that part.